tests/acceptance/test_mender-connect.py

```diff
@@ -0,0 +1,349 @@
+        )
+        connection.run(f"bash {self._sh_file}")
+        result = connection.run(f"cat {self._pid_file}")
+
+        self._pid = int(result.stdout)
+
+    def _kill(self):
+        return self._conn.run(f"kill {self._pid}", warn=True)
+
+    def kill(self):
+        """Kills the zombie process, expect to exist"""
+        result = self._kill()
+        assert result.exited == 0
+
+    def terminate(self):
+        """Kills the zombie process, do not error if it was already killed"""
+        self._kill()
+        self._conn.run(f"rm -f {self._sh_file} {self._pid_file} {self._log_file}")
+
+
+def start_mock_dbus_server(connection):
+    return SshZombieProccess(
+        connection, "python3 /tmp/mock_dbus_server.py", "mock_dbus_server"
+    )
+
+
+def wait_for_mock_dbus_server(connection):
+    start = time.time()
+    while time.time() < start + 60:
+        result = connection.run(
+            "dbus-send --print-reply --system "
+            "--dest=io.mender.AuthenticationManager "
+            "/io/mender/AuthenticationManager "
+            "io.mender.Authentication1.FetchJwtToken ",
+            warn=True,
+        )
+
+        if result.exited == 0 and "boolean true" in result.stdout:
+            # The interface is ready
+            break
+        elif (
+            "The name io.mender.AuthenticationManager was not provided by any .service files"
+            in result.stderr
+        ):
+            # The interface is not ready, sleep and retry
+            time.sleep(2)
+        else:
+            # Unexpected error, fail here
+            pytest.fail("Unknown error '%s'" % result.stderr)
+    else:
+        pytest.fail(
+            "Timed out waiting for io.mender.AuthenticationManager D-Bus interface"
+        )
+
+
+def start_mock_websockets_server(connection, port):
+    return SshZombieProccess(
+        connection,
+        f"python3 /tmp/mock_websocket_server.py {port}",
+        f"mock_websocket_server_{port}",
+    )
+
+
+def dbus_set_token_and_url(connection, token, server_url):
+    connection.run(
+        "dbus-send --print-reply --system "
+        "--dest=io.mender.AuthenticationManager "
+        "/io/mender/AuthenticationManager "
+        "io.mender.Authentication1.MockSetJwtToken "
+        f"string:'{token}' string:'{server_url}'"
+    )
+
+
+def dbus_set_token_and_url_and_emit_signal(connection, token, server_url):
+    connection.run(
+        "dbus-send --print-reply --system "
+        "--dest=io.mender.AuthenticationManager "
+        "/io/mender/AuthenticationManager "
+        "io.mender.Authentication1.MockSetJwtTokenAndEmitSignal "
+        f"string:'{token}' string:'{server_url}'"
+    )
+
+
+def wait_for_string_in_log(connection, since, timeout, search_string):
+    output = ""
+    while time.time() < since + timeout:
+        print("Searching for '%s' in mender-connect's journal:" % search_string)
+        output = connection.run(
+            "journalctl -u mender-connect -S '%s'"
```

Comment:
Why not use the long option for readability :pray:

Reply:
Okey. I'll amend [test_update_control](https://github.com/mendersoftware/meta-mender/blob/master/tests/acceptance/test_update_control.py#L681) too for consistency.